app/server/utils/ussd/directory_listing_processor.py

```diff
@@ -1,39 +1,35 @@
 import threading
+from sqlalchemy import text
 
-from server import message_processor
+from server import message_processor, db
 from server.utils.i18n import i18n_for
 from server.models.user import User
-from server.models.custom_attribute_user_storage import CustomAttributeUserStorage
 from server.models.transfer_usage import TransferUsage
 from server.models.transfer_account import TransferAccount
+from server.utils.user import default_token
 
 
 def user_directory_listing(user):
-    # TODO: add custom attribute bio
-    return f"{user.phone}"
+    bio = user.custom_attributes.filter_by(name='bio').first()
+    if bio is None:
+        return user.phone
```

Comment:
does this need to be stringified to work like in the above code?

Reply:
nope because we just stick the returned content in a formatted string, but you're right i should make it a string for consistency

Reply:
oh wait, user.phone is already a string lol. but i've added type checking now to ensure it